src/openforms/prefill/base.py

```diff
@@ -1,14 +1,22 @@
-from typing import Any, Dict, Iterable, List, Optional, Tuple
+from typing import Any, Container, Dict, Iterable, List, Optional, Tuple
 
 from openforms.authentication.constants import AuthAttribute
 from openforms.plugins.plugin import AbstractBasePlugin
 from openforms.submissions.models import Submission
+from openforms.typing import JSONEncodable
 
 from .constants import IdentifierRoles
 
 
+class AllComponentTypes(Container["BasePlugin"]):
+    def __contains__(self, thing):
+        # even the ones that don't exist
+        return True
+
+
 class BasePlugin(AbstractBasePlugin):
     requires_auth: AuthAttribute | None = None
+    for_components: Container[str] = AllComponentTypes()
```

Comment:
does this type check? Right now I read this as:

1. AllComponentTypes is a container of plugin instances (due to the `BasePlugin` type argument)
2. for_components must be a container of strings

but `BasePlugin` is not a string, so this wouldn't hold up, would it?

Reply:
Right. It ought to be a `NewType("ComponentType", str)` from `openforms.formio.typing` but I doubt we can annotate the serializers in a way that that would check.